truss/remote/baseten/remote.py

```diff
@@ -78,45 +81,55 @@ def push(  # type: ignore
+    @staticmethod
+    def _get_matching_version(model_versions: List[dict], published: bool) -> dict:
+        if not published:
+            # Return the development model version.
+            try:
+                return get_dev_version_info_from_versions(model_versions)
+            except ValueError:
+                raise click.UsageError(
+                    "No development model found. Run `truss push` then try again."
                 )
-            model_version_id = model_version["id"]
-            service_url_path = f"/model_versions/{model_version_id}"
 
-        elif isinstance(model_identifier, ModelId):
-            model = self._api.get_model_by_id(model_identifier.value)
-            model_id = model["model"]["id"]
-            model_version_id = model["model"]["primary_version"]["id"]
-            service_url_path = f"/models/{model_id}"
+        # Return the production deployment version.
+        try:
+            return get_prod_version_info_from_versions(model_versions)
+        except ValueError:
+            raise click.UsageError(
+                "No production model found. Run `truss push --publish` then try again."
+            )
 
-        elif isinstance(model_identifier, ModelVersionId):
-            model_version = self._api.get_model_version_by_id(model_identifier.value)
+    @staticmethod
+    def _get_service_url_path_and_model_ids(
+        api: BasetenApi, model_identifier: ModelIdentifier, published: bool
+    ) -> Tuple[str, str, str]:
+        if isinstance(model_identifier, ModelVersionId):
+            model_version = api.get_model_version_by_id(model_identifier.value)
             model_version_id = model_version["model_version"]["id"]
             model_id = model_version["model_version"]["oracle"]["id"]
             service_url_path = f"/model_versions/{model_version_id}"
+            return service_url_path, model_id, model_version_id
 
+        # Get model versions by either model name or ID.
+        if isinstance(model_identifier, ModelName):
+            model_id, model_versions = get_model_versions_info(api, model_identifier)
+        elif isinstance(model_identifier, ModelId):
+            model_id, model_versions = get_model_versions_info_by_id(
+                api, model_identifier
+            )
         else:
+            # Model identifier is of invalid type.
             raise click.UsageError(
                 "You must either be inside of a Truss directory, or provide --model-version or --model options."
             )
 
+        model_version = BasetenRemote._get_matching_version(model_versions, published)
+        model_version_id = model_version["id"]
+        # TODO(helen): comment on why we use models endpoint instead of
+        # model_versions. If primary version changes while this is executing,
+        # will service_url_path and model_version_id be inconsistent?
+        service_url_path = f"/models/{model_id}"
```

Comment:
@squidarth we discussed using the `models` endpoint rather than `model_versions` to handle the edge case where a user pushes a truss while this is being executed. However, in this case would the returned `model_version_id` be inconsistent with `service_url_path`? What are the implications of this?